### DISCUSSION TITLE
Installation guide

### DISCUSSION THREAD

**Discussion by facepaa** [TheSuperHackers/GeneralsGamePatch#2348] [Q&A] [answered]
Hello, is there any guide which would help me with installation of this patch? :-)
It has been a year and I still couldnt find anything :-/

**Comment by xezon** [comment]
Hello.

Clone the repository and then run the `Patch104pZH/Scripts/BuildInstallRun.bat` script on Windows. That should build, install and run it, if you have a regular game install.

If it does not work, please let us know.

**Reply by facepaa** [reply]
Hello, I got CC generals ZH from deluxe edition with gentools installed. It writes me this, see attachment. Appreciate your help!

Should I move the folder or some files into folder with game or something like that?

![ccgzh](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/135540456/702f8e93-3401-42f6-923a-122ccbad7491)

**Reply by xezon** [reply]
Interesting. So the ssl connection is rejected on your system.

See:
https://stackoverflow.com/questions/54938026/curl-unknown-error-0x80092012-the-revocation-function-was-unable-to-check-r

Do you have some sort of firewall running that blocks this https?

**Reply by facepaa** [reply, marked as answer]
Ok, got it. I just turned my antivirus off and the process started to work. Thanks for help and also for this patch you are working on! :-)

**Reply by AoKishuba** [reply]
> Hello.
> 
> Clone the repository and then run the `Patch104pZH/Scripts/BuildInstallRun.bat` script on Windows. That should build, install and run it, if you have a regular game install.
> 
> If it does not work, please let us know.

I will second the question about needing to extract to any particular folder. I ran Patch104pZH/Scripts/BuildInstallRun.bat, which appeared to run without errors, closed itself, and opened the game. The only problem is that the game launched as vanilla.

No other mods, clean install through EA app. Only thing I could think of would be that I just extracted the code to the downloads folder because no location was specified and I wanted to keep my install folder clean if possible.

**Reply by xezon** [reply]
Project source files can be copied anywhere. The fact that the game launched means that the game was found. While the game runs through Mod Builder, do you see the various Patch big files in the game install directory or are they missing? If they are missing, then something went wrong. Make sure to run Mod Builder scripts as Administrator so it has privileges to copy files into the game install directory.

If you cannot get it to work, please run BuildInstallRun.bat in Command Prompt as `BuildInstallRun.bat > log.txt` and then copy the log file here.

**Reply by AoKishuba** [reply]
No, the various SuperPatch .big files are not in C:\Program Files\EA Games.

I tried running Command Prompt as an administrator, then entering the command
C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\GeneralsGamePatch-main\Patch104pZH\Scripts\BuildInstallRun.bat > log.txt
(directory copied from Windows Explorer)

This resulted in a delay of a few seconds (not nearly as long as running the script through Windows Explorer), and then the game launched (vanilla). I cannot find "log.txt" in the patch download folder or in C:\Program Files\EA Games.

**Reply by xezon** [reply]
Ok try
```
cd /D "C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\GeneralsGamePatch-main\Patch104pZH\Scripts"
```
then
```
BuildInstallRun.bat > log.txt
```

This should place the log.txt inside the Scripts folder.

**Reply by AoKishuba** [reply]
> No, the various SuperPatch .big files are not in C:\Program Files\EA Games.
> 
> I tried running Command Prompt as an administrator, then entering the command C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\GeneralsGamePatch-main\Patch104pZH\Scripts\BuildInstallRun.bat > log.txt (directory copied from Windows Explorer)
> 
> This resulted in a delay of a few seconds (not nearly as long as running the script through Windows Explorer), and then the game launched (vanilla). I cannot find "log.txt" in the patch download folder or in C:\Program Files\EA Games.

The directory name seemed long (redundant "GeneralsGamepatch-main" nested folders from the .zip extraction process). I shortened the directory, and running "C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\Scripts\BuildInstallRun.bat > log.txt" resulted in... something. I've attached the end of the Command Prompt output here. Still no SuperPatch .big files in the game folder.

Still can't find a log.txt file in either folder.
[2024-07-20-1156 Run Attempt Output.txt](https://github.com/user-attachments/files/16321588/2024-07-20-1156.Run.Attempt.Output.txt)

**Reply by AoKishuba** [reply]
> Ok try
> 
> ```
> cd /D "C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\GeneralsGamePatch-main\Patch104pZH\Scripts"
> ```
> 
> then
> 
> ```
> BuildInstallRun.bat > log.txt
> ```
> 
> This should place the log.txt inside the Scripts folder.

(I used the shortened directory mentioned in my last reply--sorry, looks like we may have posted at the same time there)
[log.txt](https://github.com/user-attachments/files/16321601/log.txt)

No SuperPatch .big files in game folder.

**Reply by xezon** [reply]
Log looks ok. It does install (symlink) the files:

```
Copy files for InstallBundlePack_FullEnglish ...
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_899_SuperPatch_OptionalAudio.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_899_SuperPatch_OptionalAudio.big
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_899_SuperPatch_OptionalAudioEnglish.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_899_SuperPatch_OptionalAudioEnglish.big
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_899_SuperPatch_OptionalINI.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_899_SuperPatch_OptionalINI.big
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_899_SuperPatch_OptionalLangEnglish.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_899_SuperPatch_OptionalLangEnglish.big
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_899_SuperPatch_OptionalTextures.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_899_SuperPatch_OptionalTextures.big
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_899_SuperPatch_OptionalW3D.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_899_SuperPatch_OptionalW3D.big
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_900_SuperPatch_CoreAudio.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreAudio.big
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_900_SuperPatch_CoreAudioEnglish.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreAudioEnglish.big
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_900_SuperPatch_CoreINI.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreINI.big
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_900_SuperPatch_CoreLangEnglish.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreLangEnglish.big
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_900_SuperPatch_CoreMaps.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreMaps.big
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\Data\Cursors\SCCAttack7.ani
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\Data\Cursors\SCCAttack7.ani
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\English\ReadMe.txt
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\ReadMe.txt
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\English\Changes\v1.0\AllSortedByDate.md
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\Changes\v1.0\AllSortedByDate.md
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\English\Changes\v1.0\AllSortedBySeverity.md
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\Changes\v1.0\AllSortedBySeverity.md
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\English\Changes\v1.0\ArtOnlySortedByFaction.md
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\Changes\v1.0\ArtOnlySortedByFaction.md
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\English\Changes\v1.0\ChinaOnlySortedByDate.md
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\Changes\v1.0\ChinaOnlySortedByDate.md
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\English\Changes\v1.0\ControversialOnlySortedByFaction.md
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\Changes\v1.0\ControversialOnlySortedByFaction.md
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\English\Changes\v1.0\GlaOnlySortedByDate.md
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\Changes\v1.0\GlaOnlySortedByDate.md
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\English\Changes\v1.0\UsaOnlySortedByDate.md
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\Changes\v1.0\UsaOnlySortedByDate.md
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_900_SuperPatch_CoreTextures.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreTextures.big
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_900_SuperPatch_CoreW3D.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreW3D.big
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_900_SuperPatch_CoreWindow.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreWindow.big
Link C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\RawBundlePacks\FullEnglish\600_901_SuperPatch_RecoveredTextures.big
  to C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_901_SuperPatch_RecoveredTextures.big
Copy files for InstallBundlePack_FullEnglish completed in 0.035 s
```

Then it runs game

```
Run C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\generals.exe -win -quickstart
```

Is this the expected location? It appears to be a Steam or EA App game install. I have never tested it with that, but I would expect it will work.

**Reply by AoKishuba** [reply]
The install is through the EA app, and yes, that is the expected location. However, the files do not appear.

I'm wondering about the end of the full log I posted above:

> Do Uninstall ...
Revert Installed Things ...
Read pickle C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\InstalledThings.pickle ...
Remove files for InstallBundlePack_FullEnglish ...
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_899_SuperPatch_OptionalAudio.big
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_899_SuperPatch_OptionalAudioEnglish.big
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_899_SuperPatch_OptionalINI.big
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_899_SuperPatch_OptionalLangEnglish.big
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_899_SuperPatch_OptionalTextures.big
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_899_SuperPatch_OptionalW3D.big
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreAudio.big
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreAudioEnglish.big
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreINI.big
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreLangEnglish.big
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreMaps.big
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\Data\Cursors\SCCAttack7.ani
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\ReadMe.txt
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\Changes\v1.0\AllSortedByDate.md
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\Changes\v1.0\AllSortedBySeverity.md
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\Changes\v1.0\ArtOnlySortedByFaction.md
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\Changes\v1.0\ChinaOnlySortedByDate.md
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\Changes\v1.0\ControversialOnlySortedByFaction.md
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\Changes\v1.0\GlaOnlySortedByDate.md
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\English\Changes\v1.0\UsaOnlySortedByDate.md
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreTextures.big
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreW3D.big
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_900_SuperPatch_CoreWindow.big
Remove C:\Program Files\EA Games\Command and Conquer Generals Zero Hour\600_901_SuperPatch_RecoveredTextures.big
Remove files for InstallBundlePack_FullEnglish completed in 0.011 s
Remove C:\Users\LENOVO\Downloads\GeneralsGamePatch-main\Patch104pZH\.Build\InstalledThings.pickle
Revert Installed Things completed in 0.012 s

**Reply by xezon** [reply]
The way this works is, it copies the files to the game directory, then launches the game, and after the game is closed, it will remove the copies, to preserve a clean game install. According to your log, this works as expected.

In the game menu you should be able to see the Patch104p banner.

**Reply by AoKishuba** [reply]
![zhmenu](https://github.com/user-attachments/assets/c15b09ef-1d14-4ac2-929e-52d21888e634)

No banner, USA Sentry drones do not cost same as Humvees, Firebases use the Patriot sound (the two changes which seemed the quickest to check for ingame).

I watched the ZH install folder while running the script, and the SuperPatch files do indeed appear in their proper places. However, they disappear before the EA launcher initializes. With an EA app install, what happens is

1. SuperPatch files appear
2. Generals.exe triggered, but does not appear under Processes in Task manager
3. SuperPatch files disappear
4. EA App opens
5. EA App popup window informing user that game needs administrator privileges (this is _not_ the Windows popup and is not fixed by editing .exe properties to always launch in administrator mode, for any of the Generals.exe or the EA app itself)
5a. User clicks "OK"
6. Generals appears under Processes in Task Manager
7. Small screen appears for choosing between Generals and Zero Hour
7a. User clicks Zero Hour
8. Generals disappears from Processes
9. Game.dat appears under Processes
10. Game launches (as vanilla)

**Reply by xezon** [reply]
Actually, given your description, on second thought it looks like the problem here is that your generals.exe is a launcher for EA App and will trampoline to game.dat

What you can do is:

1. Open Patch104pZH\Scripts\Windows\WindowsRunner.json
2. Change `"gameExeFile": "generals.exe"` to `"gameExeFile": "game.dat"`
3. Save file

Then try again.

**Comment by AoKishuba** [comment]
Yes, that fixed it. The game now opens directly, bypassing the EA app
altogether, and loads 1.04+.

Thanks a bunch!
